addons/website/models/website.py

```diff
@@ -2076,68 +2118,38 @@ def _trigram_enumerate_words(self, search_details, search, limit):
-                    )
-                for field in direct_fields
-            ]
-            indirect_similarities = []
-            for field_info in indirect_fields.values():
-                direct = field_info['direct']
-                direct_field = model._fields[direct]
-                comodel = field_info['comodel']
-                coalias = query.make_alias(model._table, direct)
-                cofield = field_info['cofield']
-                if cofield:
-                    # One2many's comodel references the model's id.
-                    query.add_join('LEFT JOIN', coalias, comodel._table, SQL("%s = %s",
-                        SQL.identifier(model._table, 'id'),
-                        SQL.identifier(coalias, cofield),
-                    ))
-                elif 'relation' in dir(direct_field):
-                    # Many2many's relation holds the model's id in column1 and
-                    # the comodel's record id in column2.
-                    rel_alias = coalias
-                    query.add_join('LEFT JOIN', rel_alias, direct_field.relation, SQL("%s = %s",
-                        SQL.identifier(model._table, 'id'),
-                        SQL.identifier(rel_alias, direct_field.column1),
-                    ))
-                    coalias = query.make_alias(coalias, direct_field.column2)
-                    query.add_join('LEFT JOIN', coalias, comodel._table, SQL("%s = %s",
-                        SQL.identifier(rel_alias, direct_field.column2),
-                        SQL.identifier(coalias, 'id'),
-                    ))
-                indirect_similarities.append(SQL("word_similarity(%(search)s, %(field)s)",
-                    search=unaccent(SQL('%s', search)),
-                    field=unaccent(comodel._field_to_sql(coalias, field_info['indirect'], query)),
-                ))
-            similarities.extend(indirect_similarities)
-            best_similarity = SQL('GREATEST(%(similarities)s)', similarities=SQL(', ').join(similarities))
-
-            # Filter unpublished records for portal and public user for
-            # performance.
-            # TODO: Same for `active` field?
-            filter_is_published = (
-                'is_published' in model._fields
-                and model._fields['is_published'].base_field.model_name == model_name
-                and not self.env.user._is_internal()
-            )
-            if filter_is_published:
-                query.add_where('is_published')
-
-            query.order = '_best_similarity desc'
-            query.limit = 1000
-            self.env.cr.execute(query.select(
-                SQL.identifier(model._table, 'id'),
-                SQL('%s AS _best_similarity', best_similarity),
-            ))
-            ids = {row[0] for row in self.env.cr.fetchall() if row[1] and row[1] >= similarity_threshold}
+            # Group indirect_fields by comodel
+            indirect_fields_info = defaultdict(dict)
```

Comment:
nitpick: it's always nice to have a little comment next to defaultdict declaration that explains what is the "type"/"structure" of the dict. Helps with readability.
If I understand correctly, it should be a: `{comodel: {field_name: field}}` right?